frame/staking/src/pallet/mod.rs

```diff
@@ -833,15 +836,9 @@ pub mod pallet {
 			payee: RewardDestination<T::AccountId>,
 		) -> DispatchResult {
 			let stash = ensure_signed(origin)?;
-			let controller_to_be_deprecated = stash.clone();
 
-			if <Bonded<T>>::contains_key(&stash) {
-				return Err(Error::<T>::AlreadyBonded.into())
-			}
-
-			if <Ledger<T>>::contains_key(&controller_to_be_deprecated) {
-				return Err(Error::<T>::AlreadyPaired.into())
-			}
+			Self::ledger(StakingAccount::Stash(stash.clone()))
```

Comment:
I think the fact that a normal `::get()` or a getter (eg. `fn ledger`) return `Err` if something already exists is quite confusing, and I am against it. 

I think any getter, or getter-like api should behave as it should, and in fact return `Ok(_)` or `Some(_)` or default if something already exists. The code here can handle what to do if something already exists, as the previous code was doing. 

It seems like you tailore-fitted some getters for the usecases of one potential user, and I think this is not a good way to go about it.

Reply:
to nitpick, this is also slightly less efficient than just doing `contains_key` or simialr under the hood as it will decode data needlessly. Why not add a `fn is_staker` or similar to `StakingLedger`?